### PR TITLE
UCP/STREAM/TEST: Avoid returning internal endpoints form stream poll

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7024/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov pls take a look

File: src/ucp/core/ucp_ep.h [hunk 1]
```diff
@@ -42,16 +47,17 @@ enum {
                                                         (on server side due to receiving partial
                                                         worker address from the client) */
     UCP_EP_FLAG_CONNECT_PRE_REQ_QUEUED = UCS_BIT(9), /* Pre-Connection request was queued */
+    UCP_EP_FLAG_CLOSED                 = UCS_BIT(10),/* EP was closed */
```
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_ep.h]: extra flag, you can just disable `UCP_EP_FLAG_USED` flag on closure

Comment (yosefe) [reply src/ucp/core/ucp_ep.h]: the problem is that there was requirement that message which arrive *before* an internal endpoint is "retrieved" will not be dropped

Comment (evgeny-leksikov) [reply src/ucp/core/ucp_ep.h]: Then EP should be on match queue, right?

Comment (yosefe) [reply src/ucp/core/ucp_ep.h]: i disable UCP_EP_FLAG_USED on closure, it means stream recv has to drop packets if UCP_EP_FLAG_USED is not set. But it's also not set when ep is on match queue and before it was retrieved by ucp_ep_create, and we don't want this to happen

Comment (evgeny-leksikov) [reply src/ucp/core/ucp_ep.h]: forget it :) i'm thinking in scope of not implemented FIN protocol.

**Pull request merged**